app.js

```diff
@@ -1,48 +1,149 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const bodyParser = require('body-parser');
-const cookieParser = require('cookie-parser');
-const express = require('express');
-const favicon = require('serve-favicon');
-const hbs = require('hbs');
-const mongoose = require('mongoose');
-const logger = require('morgan');
-const path = require('path');
+const bodyParser = require("body-parser");
+const cookieParser = require("cookie-parser");
+const express = require("express");
+const favicon = require("serve-favicon");
+const hbs = require("hbs");
+const mongoose = require("mongoose");
+const logger = require("morgan");
+const path = require("path");
+
+// Iteration 1:
+const User = require("./models/User.model");
+const passport = require("passport");
+const session = require("express-session");
+const bcrypt = require("bcrypt");
+const LocalStrategy = require("passport-local").Strategy;
+const flash = require("connect-flash");
 
 mongoose
-  .connect('mongodb://localhost/auth-with-passport', {
+  .connect("mongodb://localhost/auth-with-passport", {
     useNewUrlParser: true,
     useUnifiedTopology: true,
-    useCreateIndex: true
+    useCreateIndex: true,
   })
-  .then(x => console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`))
-  .catch(err => console.error('Error connecting to mongo', err));
+  .then((x) =>
+    console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`)
+  )
+  .catch((err) => console.error("Error connecting to mongo", err));
 
-const app_name = require('./package.json').name;
-const debug = require('debug')(`${app_name}:${path.basename(__filename).split('.')[0]}`);
+const app_name = require("./package.json").name;
+const debug = require("debug")(
+  `${app_name}:${path.basename(__filename).split(".")[0]}`
+);
 
 const app = express();
 
 // Middleware Setup
-app.use(logger('dev'));
+app.use(logger("dev"));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 
 // Express View engine setup
 
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'hbs');
-app.use(express.static(path.join(__dirname, 'public')));
-app.use(favicon(path.join(__dirname, 'public', 'images', 'favicon.ico')));
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "hbs");
+app.use(express.static(path.join(__dirname, "public")));
+app.use(favicon(path.join(__dirname, "public", "images", "favicon.ico")));
 
 // default value for title local
-app.locals.title = 'Express - Generated with IronGenerator';
+app.locals.title = "Express - Generated with IronGenerator";
+
+// passport:
+app.use(
+  session({
+    secret: "pLeAsE-EnTeR-sEcReT",
```

Comment:
This is hillarious! xD

For future reference: You should add a SESSION_SECRET to your .env file even though it is not too important here, it will be in other cases in the furture as everything in your app.js will be available to the public.
You can reference the SESSION_SECRET in your .env file with this line:
    secret: process.env.SESSION_SECRET,